### PR TITLE
WithFiles shouldn't accept multiple files with the same paths

### PR DESCRIPTION
**Pull request by lukas-lansky**
The ticket is pretty straightforward. I'm working on checking the amount of ZIPs in production that have the property we now forbid so that we can be reasonably sure we won't break the world.

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/JobSender/Payloads/AdhocPayload.cs [hunk 1]
```diff
@@ -16,6 +18,11 @@ public AdhocPayload(string[] files)
 
         public async Task<string> UploadAsync(IBlobContainer payloadContainer, Action<string> log)
         {
+            if (FindFileNameDuplicate(out var duplicateName))
```
[the remark this answers is not in the document]
Comment (lukas-lansky) [reply src/Microsoft.DotNet.Helix/JobSender/Payloads/AdhocPayload.cs]: Failing early is certainly much better than failing later. What stopped me from this initially was my general prejudice against putting computation into constructors. I'm inclined now to move it there, let me just wait for the rest of the code review.

[the remark this answers is not in the document]
Comment (alexperovich) [reply src/Microsoft.DotNet.Helix/JobSender/Payloads/AdhocPayload.cs]: I agree, this should be in the constructor. Its not really that much computation, and it falls into the bucket of "parameter validation." Checking in the constructor and throwing an ArgumentException would be better.

[the remark this answers is not in the document]
Comment (lukas-lansky) [reply src/Microsoft.DotNet.Helix/JobSender/Payloads/AdhocPayload.cs]: Oh well, I noticed we are not using domain-specific exceptions in the codebase. :-) Moving to the constructor and removing the exception. Adding few simple tests.

File: src/Microsoft.DotNet.Helix/JobSender/Helpers.cs [hunk 2]
```diff
@@ -1,4 +1,7 @@
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.DotNet.Helix.JobSender.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
```
Comment (premun) [inline src/Microsoft.DotNet.Helix/JobSender/Helpers.cs]: QQ: Why was this needed?

Comment (lukas-lansky) [reply src/Microsoft.DotNet.Helix/JobSender/Helpers.cs]: `AdhocPayload` is internal.
